gateway/src/resty/resolver.lua

```diff
@@ -131,12 +131,15 @@ function _M.parse_nameservers(path)
     -- see https://github.com/3scale/apicast/issues/321 for more details
     insert(nameservers, resolver)
   end
-
-  for server in gmatch(resolv_conf, 'nameserver%s+([^%s]+)') do
-    -- TODO: implement port matching based on https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=549190
-    if server ~= resolver then
-      insert(nameservers, nameserver.new(server))
-    end
+  
+  for line in gmatch(resolv_conf, '(.*)\n]') do
```

Comment:
I think this could be better as:
```lua
for _,line in ipairs(re.split(resolv_conf, [[\n+$]])) do
```

Reply:
And actually I think it would be nice to extract parsing of the file into own function so it can be unit tested. I can do that if you want. 

Also looks like there is similar bug on line 116 in this file. `search` would also be extracted from a comment line.

Reply:
@mikz go ahead this was just me playing around with lua regex. But lua was winning me :)

Reply:
If you want I'm happy to provide guidance and let you do it ;)

Reply:
yay! Thanks @mikz I will do it then